OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,475 @@
+                if (UnityEngine.Input.GetKey(positiveKey) && ModInputHandler.Instance.ShouldIgnore(positiveKey))
+                {
+                    ____value -= 1f * axisDirection;
+                }
+                if (UnityEngine.Input.GetKey(negativeKey) && ModInputHandler.Instance.ShouldIgnore(negativeKey))
+                {
+                    ____value += 1f * axisDirection;
+                }
+            }
+
+            public static void DoubleAxisUpdatePost(
+                DoubleAxisCommand __instance,
+                ref Vector2 ____value,
+                KeyCode ____keyboardXPos,
+                KeyCode ____keyboardYPos,
+                KeyCode ____keyboardXNeg,
+                KeyCode ____keyboardYNeg
+            )
+            {
+                ModInputHandler.Instance.RegisterGamesBinding(__instance);
+                if (!OWInput.UsingGamepad())
+                {
+                    if (UnityEngine.Input.GetKey(____keyboardXPos) && ModInputHandler.Instance.ShouldIgnore(____keyboardXPos))
+                    {
+                        ____value.x -= 1f;
+                    }
+                    if (UnityEngine.Input.GetKey(____keyboardXNeg) && ModInputHandler.Instance.ShouldIgnore(____keyboardXNeg))
+                    {
+                        ____value.x += 1f;
+                    }
+                    if (UnityEngine.Input.GetKey(____keyboardYPos) && ModInputHandler.Instance.ShouldIgnore(____keyboardYPos))
+                    {
+                        ____value.y -= 1f;
+                    }
+                    if (UnityEngine.Input.GetKey(____keyboardYNeg) && ModInputHandler.Instance.ShouldIgnore(____keyboardYNeg))
+                    {
+                        ____value.y += 1f;
+                    }
+                }
+            }
+        }
+
+        internal static ModInputHandler Instance { get; private set; }
+
+        private HashSet<IModInputCombination> _singlePressed = new HashSet<IModInputCombination>();
+        private Dictionary<Int64, IModInputCombination> _comboRegistry = new Dictionary<Int64, IModInputCombination>();
+        private HashSet<InputCommand> _gameBindingRegistry = new HashSet<InputCommand>();
+        private float[] _timeout = new float[_maxUsefulKey];
+        private int[] _gameBindingCounter = new int[_maxUsefulKey];
+        private ModInputCombination _lastPressed;
+        private float _lastUpdate;
+        private float _lastSingleUpdate;
+        private readonly static float _updateCooldown = 0.01f; //aka little bit less than one frame
+        private readonly static float _cooldown = 0.05f;
+        private readonly static float _tapKeep = 0.3f;
+        private readonly static float _tapDuration = 0.1f;
+        private readonly static int _minUsefulKey = 8;
+        private readonly static int _maxUsefulKey = 350;
+        private readonly static int _maxComboLength = 7;
+        private readonly static string xboxPrefix = "xbox_";
```

Comment:
The order should be: private static readonly

Reply:
done